optuna/study/study.py

```diff
@@ -45,6 +48,46 @@
 _logger = logging.get_logger(__name__)
 
 
+_T = TypeVar("_T")
+
+
+def _convert_positional_args(
+    *,
+    previous_positional_arg_names: Sequence[str],
+) -> Callable[[Callable[..., _T]], Callable[..., _T]]:
+    def decorator(func: Callable[..., _T]) -> Callable[..., _T]:
+        @wraps(func)
+        def wrapper(*args: Any, **kwargs: Any) -> _T:
+            assert len(previous_positional_arg_names) <= len(signature(func).parameters), (
```

Comment:
To check `previous_positional_arg_names` list is valid, I add signature info. of function and compare them here. This helps developers to keep mapping previous signature to new signature, but my concern is that this is just a runtime overhead for users...

Reply:
Sounds good. With this logic, my understanding is that there's a runtime overhead in each invocation to the decorator function, but we only need to perform this check once, when applying the decorator, i.e. during module import? We can do this moving this assertion to an outer scope, right below `def decorator(func ...):`.

Reply:
Thanks! Fixed in [9e6b9ad](https://github.com/optuna/optuna/pull/3270/commits/9e6b9ad3d5be6ce74be00a87e158393b1702d3c8) and it works as intended.